src/app/api/services/dot-tags/dot-tags.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
+import { CoreWebService } from 'dotcms-js';
+import { Observable } from 'rxjs';
+import { DotTag } from '@models/dot-tag';
+import { RequestMethod } from '@angular/http';
+import { map, pluck } from 'rxjs/operators';
+
+/**
+ * Provide util methods to get Tags available in the system.
+ * @export
+ * @class DotTagsService
+ */
+@Injectable()
+export class DotTagsService {
+    constructor(private coreWebService: CoreWebService) {}
+
+    /**
+     * Return Tags.
+     * @returns Observable<DotTag[]>
+     * @memberof DotTagDotTagsServicesService
+     */
+    get(name?: string): Observable<DotTag[]> {
+        return this.coreWebService
+            .requestView({
+                method: RequestMethod.Get,
+                url: `v1/tags?name=${name ? name : ''}`
```

Comment:
This means we can do a request with no name? is that the right thing to do? are we doing unnecessary extra requests?

Reply:
yes, we can do request with no name, in case the user use the dropdown in the autocomplete, instead type any letter.